fabtests/functional/msg_inject.c

```diff
@@ -52,12 +52,14 @@ static int send_msg(int sendmsg, size_t size)
 	}
 
 	if (sendmsg) {
-		ret = ft_sendmsg(ep, remote_fi_addr, size,
-				&tx_ctx, flag);
-		if (ret) {
-			FT_PRINTERR("ft_sendmsg", ret);
-			return ret;
-		}
+		do {
+			ret = ft_sendmsg(ep, remote_fi_addr, size,
+					 &tx_ctx, flag);
+			if (ret && ret != -FI_EAGAIN) {
+				FT_PRINTERR("ft_sendmsg", ret);
+				return ret;
+			}
+		} while (ret == -FI_EAGAIN);
```

Comment:
Can you check how other calls handle this?  I thought they included trying to progress the CQ as well to avoid the possibility of an infinite loop.

Reply:
Done in new revision. Thought the resulted code is very close to `ft_post_tx`, which makes me wonder why `msg_inject` has an option `-N` to specifically test `fi_sendmsg`?

Reply:
sendmsg takes a different path than send (uses different flags), so there's value in testing that the functionality is the same regardless of which code path is used.